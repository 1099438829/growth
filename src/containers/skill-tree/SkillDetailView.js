import React, { Component, PropTypes } from 'react';
import { Text, View, ScrollView } from 'react-native';
import { Card, List, ListItem } from 'react-native-elements';
import * as shortid from 'shortid';

import SKILL_TREE_DATA from './SKILL_TREE_DATA';
import AppFonts from '../../theme/fonts';

// eslint-disable-next-line import/no-unresolved
const filter = require('lodash.filter');

class SkillDetailView extends Component {
  static componentName = 'SkillDetailView';

  static propTypes = {
    skillId: PropTypes.number,
  };

  static defaultProps = {
    skillId: null,
  };

  render() {
    const skillData = filter(SKILL_TREE_DATA, { id: this.props.skillId })[0];
    let skillLinkList = null;
    if (skillData.links) {
      skillLinkList = (<View>
        <Text style={{ paddingTop: 15, paddingLeft: 15, fontSize: AppFonts.h4.size }}>扩展资料</Text>
        <List>
          {
            skillData.links.map(link => (
              <ListItem
                key={shortid.generate()}
                title={link.label}
              />
            ))
          }
        </List>
      </View>);
    }

    let skillBookList = null;
    if (skillData.books) {
      skillBookList = (<View>
        <Text style={{ paddingTop: 15, paddingLeft: 15, fontSize: AppFonts.h4.size }}>推荐书籍</Text>
        <List containerStyle={{ marginBottom: 20 }}>
          {
            skillData.books.map(link => (
              <ListItem
                key={shortid.generate()}
                title={link.label}
              />
            ))
          }
        </List>
      </View>);
    }

    let rankDescriptions = null;
    if (skillData.rankDescriptions) {
      rankDescriptions = (<View>
        <Text style={{ paddingTop: 15, paddingLeft: 15, fontSize: AppFonts.h4.size }}>技能要求</Text>
        <List containerStyle={{ marginBottom: 20 }}>
          {
            skillData.rankDescriptions.map(link => (
              <ListItem
                key={shortid.generate()}
                title={link}
              />
            ))
          }
        </List>
      </View>);
    }

    return (
      <ScrollView>
        <Card
          title={skillData.title}
        >
          <Text>{ skillData.description }</Text>
        </Card>
        {rankDescriptions}
        {skillLinkList}
        {skillBookList}
      </ScrollView>
    );
  }
}

export default SkillDetailView;
