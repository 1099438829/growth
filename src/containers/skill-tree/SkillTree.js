/* eslint-disable global-require */
/* global __DEV__, require */
import React from 'react';
import { WebView, StyleSheet, Platform } from 'react-native';
import AppSizes from '../../theme/sizes';
import AppStyles from '../../theme/styles';
import { Actions } from 'react-native-router-flux';

const styles = StyleSheet.create({
  background: {
    backgroundColor: 'transparent',
    height: AppSizes.screen.height,
    width: AppSizes.screen.width,
  },
  logo: {
    width: AppSizes.screen.width * 0.85,
    resizeMode: 'contain',
  },
  whiteText: {
    color: '#FFF',
  },
});

class SkillTree extends React.PureComponent {
  static componentName = 'SkillTree';

  constructor() {
    super();
    this.webview = null;
  }

  handleMessage = (evt: any) => {
    const message = evt.nativeEvent.data;
    const skillId = JSON.parse(message).id;
    Actions.skillDetail({ skillId });
  };

  render = () => {
    let source;
    if (__DEV__) {
      source = require('./www/index.html');
    } else {
      source = Platform.OS === 'ios' ? require('./www/index.html') : { uri: 'file:///android_asset/www/index.html' };
    }

    return (
      <WebView
        ref={(webview) => {
          this.webview = webview;
        }}
        scalesPageToFit
        startInLoadingState
        onMessage={this.handleMessage}
        source={source}
        automaticallyAdjustContentInsets={false}
        style={[AppStyles.container, styles.container]}
        injectedJavaScript=""
        onNavigationStateChange={this.onNavigationStateChange}
      />
    );
  }
}

export default SkillTree;
