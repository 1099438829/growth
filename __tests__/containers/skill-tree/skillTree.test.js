import 'react-native';
import { Actions, Scene } from 'react-native-router-flux';
import React from 'react';
import SkillTree from '../../../src/containers/skill-tree/SkillTree';
import SkillDetailView from '../../../src/containers/skill-tree/SkillDetailView';

describe('test skill tree ', () => {
  // it('test render', function () {
    // eslint-disable-next-line no-unused-vars
    // const tree = renderer.create(
    //   <SkillTree skillId={1} />,
    // );
    //
    // const treeJson = tree.toJSON();
    // expect(treeJson).toMatchSnapshot();
  // });

  it('should goto detail view', () => {
    const scenesData = (
      <Scene
        key="root"
      >
        <Scene
          key={'skillDetail'}
          title={'技能'}
          component={SkillDetailView}
        />
      </Scene>
    );
    Actions.create(scenesData);
    Actions.callback = () => { };

    const spy = jest.spyOn(Actions, 'skillDetail');
    const view = new SkillTree();

    view.handleMessage({
      nativeEvent: {
        data: JSON.stringify({
          id: '1',
        }),
      },
    });
    expect(spy).toBeCalledWith({ skillId: 1 });
  });
});
