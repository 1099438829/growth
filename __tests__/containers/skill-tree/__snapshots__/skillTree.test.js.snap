// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test skill tree  test render 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTWebView
    allowsInlineMediaPlayback={undefined}
    automaticallyAdjustContentInsets={false}
    bounces={undefined}
    contentInset={undefined}
    dataDetectorTypes={undefined}
    decelerationRate={undefined}
    injectedJavaScript=""
    mediaPlaybackRequiresUserAction={undefined}
    messagingEnabled={true}
    onLoadingError={[Function]}
    onLoadingFinish={[Function]}
    onLoadingStart={[Function]}
    onMessage={[Function]}
    onShouldStartLoadWithRequest={undefined}
    scalesPageToFit={true}
    scrollEnabled={undefined}
    source={
      Object {
        "process": [Function],
      }
    }
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        Array [
          undefined,
          Object {
            "height": 1334,
            "width": 750,
          },
        ],
        Object {
          "flex": 0,
          "height": 0,
        },
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255,255,255,0.8)",
        "flex": 1,
        "height": 100,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
    />
  </View>
</View>
`;
